source/grid/grid_generator.cc

```diff
@@ -183,10 +183,10 @@ namespace GridGenerator
           , incline_factor(data.incline_factor)
           , bias_factor(data.bias_factor)
           , edge_length(1.0)
-          , n_cells_x_0(Utilities::pow(2, refinements) * n_subdivision_x_0)
-          , n_cells_x_1(Utilities::pow(2, refinements) * n_subdivision_x_1)
-          , n_cells_x_2(Utilities::pow(2, refinements) * n_subdivision_x_2)
-          , n_cells_y(Utilities::pow(2, refinements) * n_subdivision_y)
+          , n_cells_x_0(Utilities::fixed_power<2>(refinements) * n_subdivision_x_0)
```

Comment:
You need to keep `Utilities::pow` or `std::pow` here.

This statement should calculate `2^refinements`, and not `refinements^2`.

Since this is not a `constexpr` (as `refinements` is not known at compile time), you could actually turn it into `std::pow`.

Reply:
Thanks. I'll fix it now.